ako-gateway-api/k8s/gateway_controller.go

```diff
@@ -298,6 +301,100 @@ func (c *GatewayController) SetupEventHandlers(k8sinfo k8s.K8sinformers) {
 	if c.informers.SecretInformer != nil {
 		c.informers.SecretInformer.Informer().AddEventHandler(secretEventHandler)
 	}
+
+	if lib.GetServiceType() == lib.NodePortLocal {
+		podEventHandler := cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if c.DisableSync {
+					return
+				}
+				pod := obj.(*corev1.Pod)
+				namespace, _, _ := cache.SplitMetaNamespaceKey(utils.ObjKey(pod))
+				key := utils.Pod + "/" + utils.ObjKey(pod)
+				if lib.IsNamespaceBlocked(namespace) {
+					utils.AviLog.Debugf("key : %s, msg: Pod Add event: Namespace: %s didn't qualify filter", key, namespace)
+					return
+				}
+				ok, resVer := objects.SharedResourceVerInstanceLister().Get(key)
+				if ok && resVer.(string) == pod.ResourceVersion {
+					utils.AviLog.Debugf("key : %s, msg: same resource version returning", key)
+					return
+				}
+				if _, ok := pod.GetAnnotations()[lib.NPLPodAnnotation]; !ok {
+					utils.AviLog.Warnf("key : %s, msg: 'nodeportlocal.antrea.io' annotation not found, ignoring the pod", key)
+					return
+				}
+				bkt := utils.Bkt(namespace, numWorkers)
+				c.workqueue[bkt].AddRateLimited(key)
+				utils.AviLog.Debugf("key: %s, msg: ADD", key)
+			},
+			DeleteFunc: func(obj interface{}) {
+				if c.DisableSync {
+					return
+				}
+				pod, ok := obj.(*corev1.Pod)
+				if !ok {
```

Comment:
During the delete event , will pod get deleted or only annotations will get deleted ?

Reply:
this is simply the event handler for pod. updates to annotation will not trigger this event handler.